src/extractor/guidance/mergable_road_detector.cpp

```diff
@@ -32,15 +32,17 @@ inline auto makeCheckRoadForName(const NameID name_id,
     return [name_id, &node_based_graph, &node_data_container, &name_table, &suffix_table](
         const MergableRoadDetector::MergableRoadData &road) {
         // since we filter here, we don't want any other name than the one we are looking for
-        const auto road_name =
+        const auto road_name_id =
             node_data_container
                 .GetAnnotation(node_based_graph.GetEdgeData(road.eid).annotation_data)
                 .name_id;
-        if (name_id == EMPTY_NAMEID || road_name == EMPTY_NAMEID)
+        const auto road_name = name_table.GetNameForID(road_name_id).to_string();
+        if (name_id == EMPTY_NAMEID || road_name.empty())
```

Comment:
I don't understand this - why to we
- receive the name_id
- use the name_id to fetch the road name
- and only then check if the name_id is valid?

what will happen if the name id is invalid and we call GetNameForID on it?